packages/ui/src/forms/Form/index.tsx

```diff
@@ -647,26 +648,28 @@ export const Form: React.FC<FormProps> = (props) => {
       onSubmit={(e) => void contextRef.current.submit({}, e)}
       ref={formRef}
     >
-      <FormContext.Provider value={contextRef.current}>
-        <FormWatchContext.Provider
-          value={{
-            fields,
-            ...contextRef.current,
-          }}
-        >
-          <SubmittedContext.Provider value={submitted}>
-            <InitializingContext.Provider value={!isMounted || (isMounted && initializing)}>
-              <ProcessingContext.Provider value={processing}>
-                <ModifiedContext.Provider value={modified}>
-                  <FormFieldsContext.Provider value={fieldsReducer}>
-                    {children}
-                  </FormFieldsContext.Provider>
-                </ModifiedContext.Provider>
-              </ProcessingContext.Provider>
-            </InitializingContext.Provider>
-          </SubmittedContext.Provider>
-        </FormWatchContext.Provider>
-      </FormContext.Provider>
+      <EditDepthProvider>
```

Comment:
I think that is precisely what it's used for though, really the only place we use it is to alter form field label id's so they are unique per form. (We only do this for radio and checkbox, but should probably do it for all fields – besides the point though)

Reply:
> I think that is precisely what it's used for though

Edit depth is referring to how many _documents deep_ you are within the UI, not necessarily how many _forms_ deep. I should be able to freely render forms without effecting edit depth.

Reply:
This is probably better rendered within the `DocumentInfoProvider`

Reply:
Yeah I can see the meaning behind it, where do we use it though? What is its purpose?

Reply:
Like I don't think we really _need_ it?

Reply:
It's original purpose what to achieve nested documents. But as you're pointing out, that pattern is now beginning to break down because of cases this like: https://github.com/payloadcms/payload/pull/7992#discussion_r1750628737

Reply:
I mean if you have 2 relationship fields on a form and they have a reference to the same doc, this would be true today. And the useId would be the only thing making them unique.